### DISCUSSION TITLE
Como saber as medidas usadas em um projeto?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#826] [Perguntas e Respostas]
Bem, estou fazendo a carreira de front do FreeCodeCamp e bateu umas dúvidas.  Consegui entender para que serve as classes, os componentes e algumas das duvidas que eu tinha antes, consegui entender. Mas me bateu uma OUTRA duvida. Quando eu for  reproduzir um layout, ou um desenho em HTML/CSS3 como eu sei exatamente as medidas pra usar? Ou como vejo isso sem o uso de frameworks? Ou no caso, iniciante não tem problema algum em usar frameworks para isso? E bem, atualmente ando bem  triste e desanimado, pois acho que vou ter que refazer quase a grade curricular toda do meu curso de ADS e o que provavelmente consegui salvar, foi hardware e inglês. Estou me sentindo bem mal em relação a isso e passa pela minha cabeça desistir mas como é meu primeiro semestre é algo que não quero. É "normal" acontecer isso? Minhas crises de ansiedade atrapalharam bastante durante o semestre. Obrigado.

**Comment by felipefialho** [comment]
As medidas, você quer dizer as resoluções de tela?

**Comment by ghost** [comment]
As medidas que falo é das propriedades de border, margin, como sei exatamente os pixels corretos pra usar, como sei o tamanho da fonte etc  e outra duvida que veio agora,  as classes servem só pra organizar o CSS não?

**Comment by CinthiaQueiro** [comment]
Boa tarde.

Então Bruno, o que você precisa colocar em mente é que cada caso é um caso.
Primeiro sobre o uso ou não de frameworks. Hoje existem diversos que
facilitam muito o desenvolvimento.  E não é errado utilizá- los, afinal
foram feitos justamente para que não tenhamos que a cada novo projeto
reinventar a roda. Porém é super importante aprender a montar estruturas e
objetos sem utilizá- los, para saber co fazer do zero caso não exista como
utilizar os mesmos e também aprender sobre otimização.
Sobre o desânimo,  não desanime pois isso é super normal. Na mossa área
existem coisas a aprender a todo tempo e por isso às vezes nos sentimos um
tanto perdidos.
Dica; participe de fóruns e eventos, pois através deles consegue sentir o
"calor" do mercado e pra onde seguir.

Um dia de cada vez.

Abraços.

Em 23 de nov de 2017 14:32, "brunojokubauskas" <notifications@github.com>
escreveu:

**Comment by CinthiaQueiro** [comment]
Ps: estude um pouco sobre layout responsivo. Aos poucos vai sentir o que
utilizar e quando utilizar

Em 23 de nov de 2017 14:32, "brunojokubauskas" <notifications@github.com>
escreveu:

**Comment by vicentedealencar** [comment]
Eu sou um dev sem muita aptidao pra design então eu evito ficar customizando muito estilo, tento me apoiar ao máximo em algum framework. 

O campeão da década é o https://getbootstrap.com, mas eu prefiro o https://semantic-ui.com

**Comment by notnotgabriel** [comment]
@brunojokubauskas existe uma extensão do chrome chamada [Page Ruler](https://chrome.google.com/webstore/detail/page-ruler/jlpkojjdgbllmedoapgfodplfhcbnbpn?hl=pt-BR), com ela você consegue medir (em pixels) as páginas da web em geral.
Sobre tamanho das fontes existe a extensão [WhatFont](https://chrome.google.com/webstore/detail/whatfont/jabopobgcpjmedljpbcaablpmlmfcogm).

Outra skill boa que você pode aprender, é ver as medidas em programas tipo photoshop ou gimp, instale-os e tente aprender as ferramentas de medidas deles.

Pra finalizar, tem este post que ensina as medidas importantes do CSS https://css-tricks.com/the-lengths-of-css/

[]s e desanima não.
